docs: adjust API.md templates to keep </dd> inline and prevent website break

lightning-sagar opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines and make sure PR title follows Conventional Commits specification -> https://github.com/asyncapi/generator/blob/master/CONTRIBUTING.md
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

- Adjusted main-index-dl and global-index-dl templates to ensure generated docs do not break website build
- Updated templates to ensure </dd> is not pushed to a newline
- Re-generated `api.md` by running `npm run docs`

**Related issue(s)**
<!-- If you refer to a particular issue, provide its number, othewise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The 3rd option will not automatically close the issue after the merge. --> Resolves #1737 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Improved API docs formatting with consistent Markdown inline code and removed extraneous empty lines for cleaner presentation.
  * Added a new template to render global definition lists in generated docs for clearer navigation.
  * Simplified the docs generation setup to include more templates via glob patterns, expanding included content.
  * Normalized line endings for documentation and template files for more consistent display.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

derberg commented on apps/generator/package.json:
```diff
@@ -18,7 +18,7 @@
     "test:integration:update": "npm run test:integration -- -u",
     "test:cleanup": "rimraf \"test/temp\"",
-    "docs": "jsdoc2md --partial docs/jsdoc2md-handlebars/custom-sig-name.hbs docs/jsdoc2md-handlebars/main.hbs docs/jsdoc2md-handlebars/docs.hbs docs/jsdoc2md-handlebars/header.hbs docs/jsdoc2md-handlebars/defaultvalue.hbs docs/jsdoc2md-handlebars/link.hbs docs/jsdoc2md-handlebars/params-table.hbs --files lib/generator.js > docs/api.md",
+    "docs": "jsdoc2md --partial docs/jsdoc2md-handlebars/custom-sig-name.hbs docs/jsdoc2md-handlebars/main.hbs docs/jsdoc2md-handlebars/docs.hbs docs/jsdoc2md-handlebars/header.hbs docs/jsdoc2md-handlebars/defaultvalue.hbs docs/jsdoc2md-handlebars/link.hbs docs/jsdoc2md-handlebars/params-table.hbs docs/jsdoc2md-handlebars/main-index-dl.hbs docs/jsdoc2md-handlebars/global-index-dl.hbs --files lib/ ...
```
did you have a chance to see https://github.com/asyncapi/generator/pull/1738 and improvements introduced there?

lightning-sagar replied:
Thanks 🙏
 Yes, I checked #1738 , I will update the docs script to use the glob pattern:

 ```json
 "docs": "jsdoc2md --partial \"docs/jsdoc2md-handlebars/**/*.hbs\" --files lib/generator.js > docs/api.md"
 ```

 Also added overrides for:

 ```yaml
 main-index/global-index/global-index-dl.hbs  
 main-index/module-index/module-index-dl.hbs
 ```

 to remove the `md` helper and fix the unwanted `<dd>` newlines

lightning-sagar commented:
Normalized line endings and cleaned minor whitespace, no functional changes in `api.md`.
Added `.gitattributes` rules to enforce LF endings and prevent future `^M` issues:

```bash
*.sh *.bash *.md *.hbs *.js text eol=lf
```

derberg reviewed (commented):
@lightning-sagar do you mean it is ready for final review?

lightning-sagar commented:
@derberg yes

lightning-sagar commented on apps/generator/docs/api.md:
```diff
@@ -8,15 +8,15 @@ Reference API documentation for AsyncAPI Generator library.
 <dl>
 <dt><a href="#Generator">Generator</a></dt>
-<dd></dd>
+
```
yes, you were right, the newline came from the autogenerated part, I removed the `module-index-dl.hbs` and kept only `global-index-dl.hbs`, added whitespace trimming (~)
also please do let me know if there is anything else you'd like me to update...
thanks

derberg reviewed (commented):
can you add somewhere in the comment a link or copy/paste version of oryginal [global-index-dl.hbs](https://github.com/asyncapi/generator/pull/1748/files#diff-97dc8f2694159e0c51cb314832d436118b1278f87fe55f13e1f49a472d053672) file for review?

lightning-sagar commented:
here is the original code:
`node_modules/dmd/partials/main-index/global-index/global-index-dl.hbs`:

```hbs
{{#globals kind=kind ~}}
{{#if @first~}}{{>heading-indent}}{{../title}}

<dl>
{{/if~}}
<dt>{{>sig-link-html}}</dt>
<dd>{{{md (inlineLinks description)}}}</dd>
{{#if @last~}}</dl>

{{/if~}}
{{/globals~}}
```

derberg commented on apps/generator/docs/jsdoc2md-handlebars/main-index/global-index/global-index-dl.hbs:
```diff
@@ -0,0 +1,11 @@
+<dl>
+{{/if~}}
+<dt>{{>sig-link-html}}</dt>
+{{#if description~}}<dd><p>{{{inlineLinks description}}}</p></dd>{{/if~}}
```
why did you remove `md` usage that was in original template

lightning-sagar replied:
I removed the `md` because it wraps the content in `<p>` , which introduced extra spacing and newlines inside `<dd>`, causing the generated api.md to break the website layout...
Please check the img below ( output when I added the `md` ):
<img width="1334" height="171" alt="image" src="https://github.com/user-attachments/assets/aa14f8e4-4e00-4938-b182-4485b2d5c170" />

lightning-sagar replied:
The issue was that whenever I ran `npm run docs` in my local vscode, it automatically switched the line endings to CRLF...
Just to confirm this, I set up a CI that runs `npm run docs` and then checks if there are any `diffs in the generated api.md`...
Here is the result from the CI run: 
https://github.com/lightning-sagar/generator/actions/runs/19354578957/job/55373299546

This helped confirm that the CRLF was introduced locally, and the actual content generation logic is fine

lightning-sagar commented:
@derberg , the MDX preview build issue is now fixed..
the problem was caused by a newline between </dd> and </dl> which resulted in an invalid MDX structure.... 

thank so much for cooperating with me, correcting me, and taking the time to guide contributors... really means a lot for me, thanks!
